R/gh_token.R

```diff
@@ -53,6 +53,20 @@ gh_token <- function(api_url = NULL) {
   gh_pat(token$password %||% "")
 }
 
+#' @export
+#' @rdname gh_token
+gh_token_exists <- function(api_url = NULL) {
+  api_url <- api_url %||% default_api_url()
+  tryCatch(
+    {
+      token <- gitcreds::gitcreds_get(get_hosturl(api_url))
+      gh_pat(token$password)
+      TRUE
+    } ,
+    error = function(e) FALSE
+  )
```

Comment:
Should this be factored in such a way that it can be used inside `gh_token()` just above?

Reply:
It's not immediately obvious to me how to do that due to a few subtle differences in the functions.